### DISCUSSION TITLE
ppx rewriter - current file path as the `cwd` instead of the project root

### DISCUSSION THREAD

**Discussion by anentropic** [ocamllabs/vscode-ocaml-platform#1173] [Q&A]
I made a ppx extension rewriter

It takes a string constant as an arg, representing a file path.

If I use a relative path like `[%myppx "lib/myppx/data.json"]` then I get a red underline in VS Code, with the error:
```
I/O error: lib/myppx/data.json: No such file or directory
ocamllsp
```
(this code would be in say `lib/myproject/some.ml`)

However I can compile and run my code via `dune` no problem.

VS Code project was opened from my project root.

So the issue seems to be that ocamllsp doesn't have the project root as the `cwd` ?

If I change the path to `[%myppx "../myppx/data.json"]` then ocamllsp is happy, but I can no longer build in `dune`. This implies ocamllsp is using the current file path as the `cwd` instead of the project root.

Is there a config setting I can use to fix this?

**Comment by anentropic** [comment]
I tried adding `root` to `.vscode/settings.json`:
```json
    "ocaml.sandbox": {
        "kind": "opam",
        "switch": "${workspaceFolder:my-project}",
        "root": "/Users/anentropic/Documents/Dev/my-project-root"
    },
```
but it didn't help, even after closing and re-opening the IDE

**Comment by smorimoto** [comment]
The workspace might help with that: https://code.visualstudio.com/docs/editor/workspaces
